### PR TITLE
feat: support all the config resolving related babel options

### PR DESCRIPTION
**Pull request by jquense**
I realized trying to get this to work with gatsby in a monorepo that react-docgen doesn't have enough info to resolve my local configs and babelrc

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -362,18 +359,18 @@ we are getting this output:
...
+| Tuple                                     | `let x: [foo, "value", number];`                                                                                                                                        | `{ "name": "tuple", "raw": "<raw-signature>", "elements": [{ <element-type> }, ...] }`                                                                                                                                                                             |
+| Union                                     | `let x: number | string;`                                                                                                                                               | `{ "name": "union", "raw": "<raw-signature>", "elements": [{ <element-type> }, ...] }`                                                                                                                                                                             |
```
Comment (danez) [inline README.md]: Can we escape the pipe in the union somehow or use an escape sequence?

It is currently treated as table separator.

**Comment by danez** [comment]
maybe this will also help @fkling with this problem https://github.com/reactjs/react-docgen/commit/60196fc30f6ed9ff79e0c241edc10ba4fc4758cc#comments?

File: README.md (hunk 1 above)
Comment (jquense) [reply README.md]: bah sorry not sure what's happening thing, maybe prettier being over zealous...i'll fix

**Comment by jquense** [comment]
ok readme fixed 👍

**Comment by fkling** [comment]
So, I figured out what the issue is: The CI environment has a babel config file in its root that gets picked up. However, the necessary packages are not installed (and I can't get them to).

I'd like to disable loading the babel config file from the file system and I thought `babelrc: false` would do that, but it doesn't seem to work? Am I missing something here?

**Comment by danez** [comment]
> Project-wide configs can also be disabled by setting "configFile" to false.

If you use `babel.config.js` this applies.

https://babeljs.io/docs/en/config-files

**Comment by fkling** [comment]
Thanks, I have figured that out now as well 🤣  So many options...

**Comment by fkling** [comment]
Not sure if ignoring local config files is a common enough issue to have a command line flag for...

**Pull request merged**